src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -1261,6 +1261,13 @@ std::pair< bool, nuraft::cb_func::ReturnCode > RaftReplDev::handle_raft_event(nu
                 auto req = m_state_machine->localize_journal_entry_prepare(*entry);
                 if (req == nullptr) {
                     sisl::VectorPool< repl_req_ptr_t >::free(reqs);
+		    // The hint set here will be used by the next after next appendEntry, the next one
+		    // always go with -1 from NuRraft code.
+		    //
+                    // We are rejecting this log entry, meaning we can accept previous log entries.
+		    // If there is nothing we can accept(i==0), that maens we are waiting for commit
+		    // of previous lsn, set it to 1 in this case.
+                    m_state_machine->reset_next_batch_size_hint(std::max(1ul, i));
```

Comment:
Can you help me understand if we are setting 1 here, how can we get `batch_size_hint_cnt` to be set as -1  at `HomeRaftLogStore::log_entries_ext`, so that we can apply empty append entries and only send commit to the follower?

Reply:
https://github.com/eBay/NuRaft/blob/1adcc6282109c2ddf1121bbc374d48d303145e39/src/handle_append_entries.cxx#L703